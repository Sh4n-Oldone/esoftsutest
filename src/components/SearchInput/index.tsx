import React from 'react';
import {StyleSheet, TextInput} from 'react-native';

const SearchInput: React.FC<{
  textHandler: (text: string) => void;
  inputValue: string;
}> = ({textHandler, inputValue}) => {
  return (
    <TextInput
      style={styles.input}
      onChangeText={textHandler}
      value={inputValue}
      placeholder='Используйте английскую раскладку'
    />
  );
};

const styles = StyleSheet.create({
  input: {
    height: '100%',
    width: '85%',
    padding: 10,
    backgroundColor: '#fff',
  },
});

export default SearchInput;
