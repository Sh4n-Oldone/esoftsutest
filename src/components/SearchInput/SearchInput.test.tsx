import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';
import SearchInput from './index';

// const mockedNavigate = jest.fn();

// jest.mock('@react-navigation/native', () => ({
//   useNavigation: () => ({navigate: mockedNavigate}),
// }));

describe('Search input component', () => {
  it('Input rendered correctly', () => {
    const component = renderer.create(<SearchInput />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Input validate correctly', () => {
    const onChangeTextMock = jest.fn(text => text.replace(/[^A-Za-z-]/g, ''));
    const newTestValue = '123cats123';
    const input = renderer.create(
      <SearchInput
        textHandler={onChangeTextMock}
        inputValue={onChangeTextMock(newTestValue)}
      />,
    );
    expect(input.toJSON().props.value).toBe('cats');
  });
});
