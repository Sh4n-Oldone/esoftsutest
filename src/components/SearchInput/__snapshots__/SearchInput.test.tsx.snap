// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search input component Input rendered correctly 1`] = `
<TextInput
  placeholder="Используйте английскую раскладку"
  style={
    Object {
      "backgroundColor": "#fff",
      "height": "100%",
      "padding": 10,
      "width": "85%",
    }
  }
/>
`;
